app/agents/RecruitmentStageManagerAgent.py

```diff
@@ -0,0 +1,213 @@
+            "request",
+            "start_request",
+            MessageType.START_REQUEST,
+            [f"{self.agent.jid}", f"{self.agent.recruitment_stage.priority}"],
+        )
+
+        self.agent.logger.info("Sending message to rm agent with the request to start.")
+        await self.send(msg)
+
+        msg = await self.receive(timeout=10)
+        if msg is None:
+            return
+
+        _, data = await self.agent.get_message_type_and_data(msg)
+        start_allowed = await self.evaluate_stage_start(data[0])
+        if start_allowed:
+            self.agent.track_stage_behav = TrackStage(period=TRACK_STAGE_PERIOD)
+            self.agent.add_behaviour(self.agent.track_stage_behav)
+
+            self.kill()
+
+    async def evaluate_stage_start(self, body: str) -> bool:
+        start_allowed = True if body == "True" else False
+        self.agent.logger.info(
+            f"Received message from rm agent with permission to start: {start_allowed}."
+        )
+
+        if (
+            start_allowed
+            and self.agent.recruitment_stage.status == RecruitmentStageStatus.CREATED
+        ):
+            self.agent.recruitment_stage.status = RecruitmentStageStatus.IN_PROGRESS
+            self.agent.recruitment_stage_module.update(
+                self.agent.recruitment_stage._id, {"status": 2}
+            )
+
+        return start_allowed
+
+
+class TrackStage(spade.behaviour.PeriodicBehaviour):
+    """Behaviour representing TrackStageProgress activity and SendStageResult protocol in order to send stage result when status is DONE"""
+
+    agent: RecruitmentStageManagerAgent
+
+    async def run(self):
+        self.agent.logger.info("TrackStage behaviour run.")
+
+        await self.check_stage_progress()
+
+    async def check_stage_progress(self):
+        self.agent.recruitment_stage = self.agent.recruitment_stage_module.get(
+            self.agent.recruitment_stage._id
+        )
+
+        if self.agent.recruitment_stage.status == RecruitmentStageStatus.DONE:
+            self.agent.logger.info("Stage has been completed.")
+            await self.send_and_receive()
+        else:
+            self.agent.logger.info("Stage is in progress.")
+
+    async def send_and_receive(self):
+        msg = await self.agent.prepare_message(
+            self.agent.rm_jid,
+            "inform",
+            "stage_result",
+            MessageType.STAGE_RESULT,
+            [f"{self.agent.jid}", f"{self.agent.recruitment_stage.result}"],
+        )
+
+        self.agent.logger.info("Sending message to rm agent with the stage result.")
+        await self.send(msg)
+
+        msg = await self.receive(timeout=10)
+        if msg is None:
+            return
+
+        _, data = await self.agent.get_message_type_and_data(msg)
+        if data[0] == "ACK":
+            self.agent.logger.info(
+                "Rment agent received stage result. Exiting agent..."
+            )
+            await self.agent.stop()
```

Comment:
pytanie, nie uwaga - czy to jest potrzebne? bo teoretycznie jak wysylamy inform to nas to nie obchodzi tak? tylko to jest po to zeby ladnie zamknąć komunikację?
+ czy w przypadku gdy jakimś cudem Rment nie odbierze to ten behav ponowi tę próbę? testowałeś to?

Reply:
imo to wazne jest , by miec potwierdzenie odbioru, bo jak wiem, ze odebral to moge zamykac agenta. jesli by zamknal agenta bez potwierdzenia to nie mam pewnosci, ze wiadomosc przyszla

Reply:
wiec tak, to jest po to by zamknac go, majac pewnosc ze recruitment dodal result do overall_result

Reply:
odnosnie tego czy ponowi: tak, bo to jest periodic behaviour wiec znow sprawdzi czy status jest na DONE i znow wysle swoj result